pkg/controllers/state/statenode.go

```diff
@@ -104,8 +130,9 @@ type StateNode struct {
 
 	// TODO remove this when v1alpha5 APIs are deprecated. With v1beta1 APIs Karpenter relies on the existence
```

Comment:
Nope, we actually had a discussion about this a long time ago that meant we could get rid of this comment. The TLDR is that we're keeping the MarkedForDeletion in here so that we can ensure that we mark the node as deleted rather than waiting for the existence of the taint for knowing about successive disruption actions